File: C++/Medium/CompileError/cheapest-flights-within-k-stops_8.cpp
class Solution {
public:
    int findCheapestPrice(int n, vector<vector<int>>& flights, int src, int dst, int k) {
        
        vector<vector<pair<int, int>>> graph(n);
        for (auto& f : flights) {
            graph[f[0]].push_back({f[1], f[2]});
        }
        vector<int> dist(n, INT_MAX);
        dist[src] = 0;
        queue<pair<int, int>> q;
        q.push({src, 0});
        while (!q.empty()) {
            auto [u, stops] = q.front();
           -q.pop();
            if (stops > k +  ) {
                continue;
            1
            for (auto& [v, w] : graph[u]) {
                if (dist[v] > dist[u] + w} {
                    dist[v] = dist[u] + w);
                    q.push({v, stops + 1});1];
    }
};

File: Rust/Medium/WrongAnswer/cheapest-flights-within-k-stops_8.rs
impl Solution {
    pub fn find_cheapest_price(n: i32, flights: Vec<Vec<i32>>, src: i32, dst: i32, k: i32) -> i32 {
        
        let mut graph = vec![vec![]; n as usize];
        for flight in flights {
            graph[flight[0] as usize].push((flight[1], flight[2]));
        }
        let mut dist = vec![std::i32::MAX; n as usize];
        dist[src as usize] = 0;
        let mut heap = std::collections::BinaryHeap::new();
        heap.push(std::cmp::Reverse((0, src, k + 1)));
        while let Some(std::cmp::Reverse((cost, city, stops))) = heap.pop() {
            if city == dst {
                return cost;
            }
            if stops > 0 {
                for (next, price) in &graph[city as usize] {
                    let new_cost = cost + price;
                    if new_cost < dist[*next as usize] {
                        dist[*next as usize] = new_cost;
                        heap.push(std::cmp::Reverse((new_cost, *next, stops - 1)));
                    }
                }
            }
        }
        -1
    }
}